### PR TITLE
Add toggle for custom_from_subdomain DNS record creation

### LINKED ISSUE #98: Allow managing custom subdomain DNS records outside of module and account
### Describe the Feature

A variable to toggle the `custom_from_subdomain` record creation.

### Expected Behavior

Toggle a variable `create_custom_from_subdomain_record` from it's default `true` to `false` to skip the Route53 record creation for the subdomains.

### Use Case

The module currently has no way to disable the `custom_from_domain` Route53 records from being created. This is a problem for teams that will need to manage DNS records outside of Route53 or in a different AWS account.

### Describe Ideal Solution

Add this new variable `create_custom_from_subdomain_record` and default it to `true` to preserve the current behavior for user that deployed the latest version of the module. This will then allow those that need this feature to toggle the option to `false` and get the desired results.

### Alternatives Considered

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by sarkis**
## what

This module currently has no way to disable the custom_from_domain Route53 records from being created. Add a new variable `create_custom_from_subdomain_record` and default it to `true` to preserve the current behavior. Allows users that need this feature to toggle the option to false and get the desired results.

## why

This is a problem for organizations that will need to manage DNS records outside of Route53 or in a different AWS account. Currently, other records in the module have a toggle, but the one for `custom_from_subdomain` seems to have been missed.

## references

closes #98

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -55,6 +55,12 @@ variable "custom_from_behavior_on_mx_failure" {
   }
 }
 
+variable "custom_from_dns_record_enabled" {
+  type        = bool
+  description = "If provided the module will create a Route53 DNS record for the `From` address subdomain."
```
Comment (osterman) [inline variables.tf:60]:
```suggestion
  description = "If enabled the module will create a Route53 DNS record for the `From` address subdomain."
```

**Comment by osterman** [comment]
I'm checking with the team

File: main.tf [hunk 2]
```diff
@@ -60,7 +60,7 @@ data "aws_region" "current" {
 }
 
 resource "aws_route53_record" "custom_mail_from_mx" {
-  count = local.custom_from_subdomain_enabled ? 1 : 0
+  count = local.custom_from_subdomain_enabled && custom_from_dns_record_enabled ? 1 : 0
```
Comment (Benbentwo) [inline main.tf:63]:
```suggestion
  count = local.custom_from_subdomain_enabled && var.custom_from_dns_record_enabled ? 1 : 0
```

**Review by Benbentwo — CHANGES_REQUESTED** [review]
Looks like missed `var.`

**Review by Benbentwo — APPROVED** [review]

**Review by Benbentwo — APPROVED** [review]

**Review by osterman — APPROVED** [review]

**Pull request merged**